commonmark-ext-task-list-items/src/main/java/org/commonmark/ext/task/list/items/internal/TaskListItemInlineParser.java

```diff
@@ -0,0 +1,47 @@
+package org.commonmark.ext.task.list.items.internal;
+
+import org.commonmark.ext.task.list.items.TaskListItemMarker;
+import org.commonmark.internal.InlineParserImpl;
+import org.commonmark.node.ListItem;
+import org.commonmark.node.Node;
+import org.commonmark.parser.InlineParserContext;
+
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * An extension of {@link InlineParserImpl} which parses {@link TaskListItemMarker}s in {@link ListItem}s.
+ */
+public class TaskListItemInlineParser extends InlineParserImpl {
+
+    private static final Pattern REGEX_BULLET = Pattern.compile("[\\-+*]?");
```

Comment:
I don't think this part would ever match, because by the time inline parsing is done, the list item marker is already parsed and not part of `input`.

Reply:
Looks like you're right. I've removed it from the regexp used by the new PostProcessor.